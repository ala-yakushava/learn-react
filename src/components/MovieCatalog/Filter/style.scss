@import "../../../styles/theme";

.Filter {
  display: flex;
  justify-content: space-between;
  padding: 15px 0 0;
  border-bottom: 3px solid $grey;

  &_list {
    display: flex;
    align-items: center;
  }

  &_item {
    @include reset-button;
    margin-right: 12px;
    padding: 5px 10px;
    font-size: 18px;
    line-height: 1.2;
    color: $white;
    text-transform: uppercase;

    &:focus {
      border-bottom: 1px solid red;
    }
  }

  &_select {
    display: flex;
    width: 300px;
  }

  &_label {
    display: flex;
    align-items: center;
    width: 150px;
    font-size: 18px;
    line-height: 1.2;
    color: rgb(209, 203, 203);
    text-transform: uppercase;
  }
}
