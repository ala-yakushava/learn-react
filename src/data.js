export const formField = [
  {
    name: 'title',
    label: 'Title',
    placeholder: 'Title here',
  },
  {
    name: 'release_date',
    label: 'Release date',
    placeholder: 'Select date',
  },
  {
    name: 'url',
    label: 'Movie URL',
    placeholder: 'Movie URL here',
  },
  {
    name: 'genres',
    label: 'Genre',
    placeholder: 'Select genre',
  },
  {
    name: 'overview',
    label: 'Overview',
    placeholder: 'Overview text goes here',
  },
  {
    name: 'runtime',
    label: 'Runtime',
    placeholder: 'Runtime text goes here',
  },
];

export const genres = [
  {
    id: "00",
    label: "all",
    value: "all",
  },
  {
    id: "01",
    label: "documentary",
    value: "documentary",
  },
  {
    id: "02",
    label: "comedy",
    value: "comedy",
  },
  {
    id: "03",
    label: "horror",
    value: "horror",
  },
  {
    id: "04",
    label: "crime",
    value: "crime",
  },
];

export const options = [
  {
    label: 'Title',
    value: 'Title',
  },
  {
    label: 'Release date',
    value: 'Release date',
  },
];

export const movies = [
  {
    id: "01",
    title: "Pupl Fiction",
    genres: ['Action & Adventure'],
    release_date: 2004,
    poster_path: "./public/img-1.img",
  },
  {
    id: "02",
    title: "Pupl Fiction",
    genres: ['Action & Adventure'],
    release_date: 2004,
    poster_path: "./public/img-1.img",
  },
  {
    id: "03",
    title: "Pupl Fiction",
    genres: ['Action & Adventure'],
    release_date: 2004,
    poster_path: "./public/img-1.img",
  },
  {
    id: "04",
    title: "Pupl Fiction",
    genres: ['Action & Adventure'],
    release_date: 2004,
    poster_path: "./public/img-1.img",
  },
  {
    id: "05",
    title: "Pupl Fiction",
    genres: ['Action & Adventure'],
    release_date: 2004,
    poster_path: "./public/img-1.img",
  },
  {
    id: "06",
    title: "Pupl Fiction",
    genres: ['Action & Adventure'],
    release_date: 2004,
    poster_path: "./public/img-1.img",
  },
];
