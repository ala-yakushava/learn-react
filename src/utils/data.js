import uniqueId from 'lodash/uniqueId';

export const sortItems = [
  {
    id: 'sort-rating',
    value: 'vote_average',
    label: 'Rating',
  },
  {
    id: 'sort-release_date',
    value: 'release_date',
    label: 'Release date',
  },
];

export const filterItems = [
  {
    id: 'filter-all',
    value: 'All',
    label: 'All',
  },
  {
    id: 'filter-documentary',
    value: 'Documentary',
    label: 'Documentary',
  },
  {
    id: 'filter-comedy',
    value: 'Comedy',
    label: 'Comedy',
  },
  {
    id: 'filter-horror',
    value: 'Horror',
    label: 'Horror',
  },
  {
    id: 'filter-crime',
    value: 'Crime',
    label: 'Crime',
  },
];

export const formField = [
  {
    id: uniqueId(),
    name: 'title',
    type: 'text',
    label: 'Title',
    placeholder: 'Title here',
  },
  {
    id: uniqueId(),
    name: 'release_date',
    type: 'date',
    label: 'Release date',
    placeholder: 'Select date',
  },
  {
    id: uniqueId(),
    name: 'poster_path',
    type: 'url',
    label: 'Movie URL',
    placeholder: 'Movie URL here',
  },
  {
    id: uniqueId(),
    name: 'genres',
    type: 'text',
    label: 'Genre',
    placeholder: 'Select genre',
  },
  {
    id: uniqueId(),
    name: 'overview',
    type: 'text',
    label: 'Overview',
    placeholder: 'Overview text goes here',
  },
  {
    id: uniqueId(),
    name: 'runtime',
    type: 'text',
    label: 'Runtime',
    placeholder: 'Runtime text goes here',
  },
];
