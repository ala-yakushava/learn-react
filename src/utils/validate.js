import * as yup from 'yup';

const MovieFormSchema = yup.object().shape({
  title: yup.string().transform((value) => value.trim()).min(2).required('Required'),
  release_date: yup.date().max(new Date(), 'Future date'),
  poster_path: yup.string().url('Invalid URL').required('Required'),
  genres: yup.string().transform((value) => value.trim()).min(3).required('Required'),
  overview: yup.string().transform((value) => value.trim()).min(10).required('Required'),
  runtime: yup.number('Invalid number').required('Required').min(0, 'Too Short!').integer(),
});

export default MovieFormSchema;
