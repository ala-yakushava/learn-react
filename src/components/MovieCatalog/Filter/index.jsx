import React from 'react';
import PropTypes from 'prop-types';

import './style.scss';

const Filter = ({ filterItems, defaultValue, onClick }) => {
  const cn = (value) => {
    const isActive = value === defaultValue;
    return isActive ? 'Filter_item active' : 'Filter_item';
  };

  return (
    <div className="Filter" data-testid="filter">
      { filterItems.map(({ id, value, label }) => (
        <button type="button" className={cn(value)} key={id} onClick={onClick(value)}>
          { label }
        </button>
      )) }
    </div>
  );
};

Filter.propTypes = {
  filterItems: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
      label: PropTypes.string.isRequired,
    }),
  ).isRequired,
  defaultValue: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Filter;
