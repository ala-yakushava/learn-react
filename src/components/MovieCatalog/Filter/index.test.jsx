import React from 'react';
import renderer from 'react-test-renderer';

import Filter from './index';
import { filterItems } from '../../../utils/data';

describe('Components - Filter', () => {
  test('should match snapshot', () => {
    const tree = renderer
      .create(<Filter filterItems={filterItems} defaultValue="All" onClick={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
