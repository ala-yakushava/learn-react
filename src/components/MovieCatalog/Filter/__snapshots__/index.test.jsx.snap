// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - Filter should match snapshot 1`] = `
<div
  className="Filter"
  data-testid="filter"
>
  <button
    className="Filter_item active"
    type="button"
  >
    All
  </button>
  <button
    className="Filter_item"
    type="button"
  >
    Documentary
  </button>
  <button
    className="Filter_item"
    type="button"
  >
    Comedy
  </button>
  <button
    className="Filter_item"
    type="button"
  >
    Horror
  </button>
  <button
    className="Filter_item"
    type="button"
  >
    Crime
  </button>
</div>
`;
